.github/workflows/main.yml

```diff
@@ -0,0 +1,38 @@
+name: Main
+on: [push]
+
+jobs:
+  build:
+    name: Build
+    runs-on: [self-hosted, AWS, Linux]
+    timeout-minutes: 10
+    steps:
+      - uses: Brightspace/third-party-actions@actions/checkout
+      - name: Setup .NET Core
+        uses:  Brightspace/third-party-actions@actions/setup-dotnet
+        with:
+          dotnet-version: 7.0.x
+
+      - uses: Brightspace/codeartifact-actions/nuget/add-source@main
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          AWS_SESSION_TOKEN: ${{ secrets.AWS_SESSION_TOKEN }}
+        with:
+            role-arn: arn:aws:iam::569998014834:role/github+Brightspace+D2L.Hypermedia.Siren
+  
+      - name: dotnet restore
+        run: dotnet restore
+      
+      - name: dotnet build
+        run: dotnet build --configuration Release
+
+      - name: dotnet test
+        run: dotnet test D2L.Hypermedia.Siren.Tests --configuration Release --no-build
+
+      - name: dotnet pack
+        run: dotnet pack --configuration Release --no-build
+
+      - name: Publish to CodeArtifact
+        if: github.ref == 'refs/heads/master'
+        run: dotnet nuget push "**/*.symbols.nupkg" --source d2l/private --api-key aws
```

Comment:
We could use [this](https://github.com/Brightspace/D2L.CodeStyle/blob/main/.github/workflows/build.yml#L28) instead, but I'm not sure there's a difference.